IPC_computetrigger.py

```diff
@@ -0,0 +1,189 @@
+    df["date"] = pd.to_datetime(df["date"])
+    df["year"] = df["date"].dt.year
+    df["month"] = df["date"].dt.month
+
+    df["Source"] = source
+
+    df = df.rename(columns={"adjusted_population": "pop_Country", "ADM1_EN": "ADMIN1"})
+
+    # calculate percentage of population per status and level
+    for status in ["CS", "ML1", "ML2"]:
+        for i in range(1, 6):
+            c = f"{status}_{i}"
+            df[f"perc_{c}"] = df[c] / df[f"pop_{status}"] * 100
+        # get pop and perc in IPC3+ and IPC2-
+        df[f"{status}_3p"] = df[[f"{status}_{i}" for i in range(3, 6)]].sum(axis=1)
+        df[f"perc_{status}_3p"] = df[f"{status}_3p"] / df[f"pop_{status}"] * 100
+        df[f"{status}_2m"] = df[[f"{status}_{i}" for i in range(1, 3)]].sum(axis=1)
+        df[f"perc_{status}_2m"] = df[f"{status}_2m"] / df[f"pop_{status}"] * 100
+    df["perc_inc_ML2_3p"] = df["perc_ML2_3p"] - df["perc_CS_3p"]
+    df["perc_inc_ML1_3p"] = df["perc_ML1_3p"] - df["perc_CS_3p"]
+    return df
+
+
+def get_trigger(row, status, level, perc):
+    """
+    Return 1 if percentage of population in row for status in level "level" or higher, equals or larger than perc
+    """
+    # range till 6 cause 5 is max level
+    cols = [f"{status}_{l}" for l in range(level, 6)]
+    if np.isnan(row[f"pop_{status}"]):
+        return np.nan
+    if round(row[cols].sum() / row[f"pop_{status}"] * 100) >= perc:
+        return 1
+    else:
+        return 0
+
+
+# TODO: we are not using the relative increase at the moment, do we want to remove it?
+def get_trigger_increase_rel(row, level, perc):
+    """
+    Return 1 if population in row for >="level" at ML1 is expected to be larger than (current (CS) population in >=level) * (1+(perc/100))
+    """
+    # range till 6 cause 5 is max level
+    cols_ml1 = [f"ML1_{l}" for l in range(level, 6)]
+    cols_cs = [f"CS_{l}" for l in range(level, 6)]
+    if row[["pop_CS", "pop_ML1"]].isnull().values.any():
+        return np.nan
+    elif row[cols_ml1].sum() == 0:
+        return 0
+    elif row[cols_ml1].sum() > 0 and row[cols_cs].sum() == 0:
+        return 1
+    elif (
+        round((row[cols_ml1].sum() - row[cols_cs].sum()) / row[cols_cs].sum() * 100)
+        >= perc
+    ):
+        return 1
+    else:
+        return 0
+
+
+def get_trigger_increase(row, status, level, perc):
+    """
+    Return 1 if for row percentage in >="level" at status minus percentage in >="level" currently (CS) is expected to be larger than perc
+    For Global IPC the population analysed in ML2 is sometimes different than in CS. That is why we work dirrectly with percentages and not anymore with (pop status level+ - pop CS level+) / pop CS
+    """
+    # range till 6 cause 5 is max level
+    cols_perc_ml = [f"perc_{status}_{l}" for l in range(level, 6)]
+    cols_perc_cs = [f"perc_CS_{l}" for l in range(level, 6)]
+    if row[["pop_CS", f"pop_{status}"]].isnull().values.any():
+        return np.nan
+    if row[cols_perc_ml].sum() == 0:
```

Comment:
should this specify which ML?

Reply:
The ML is specified by "status". Is that what you mean?